src/live_restore/live_restore_state.c

```diff
@@ -0,0 +1,474 @@
+
+    return (ret);
+}
+
+/*
+ * __wti_live_restore_init_state --
+ *     Initialize the live restore state. Read the state from file if it exists, otherwise we start
+ *     in the log copy state and need to create the file on disk.
+ */
+int
+__wti_live_restore_init_state(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FS *lr_fs)
+{
+    WT_DECL_RET;
+    WT_FILE_HANDLE *fh = NULL;
+
+    __wt_writelock(session, &lr_fs->state_lock);
+
+    WT_ASSERT_ALWAYS(session, lr_fs->state == WTI_LIVE_RESTORE_STATE_NONE,
+      "Attempting to initialize already initialized state!");
+
+    WT_DECL_ITEM(state_file_name);
+    WTI_LIVE_RESTORE_STATE state;
+    WT_RET(__live_restore_get_state_from_file(
+      session, lr_fs->os_file_system, lr_fs->destination.home, &state));
+
+    if (state != WTI_LIVE_RESTORE_STATE_NONE) {
+        lr_fs->state = state;
+    } else {
+        /*
+         * The state file doesn't exist which means we're starting a brand new live restore. Create
+         * the state file in the log copy state.
+         */
+        char state_to_write[128];
+        __live_restore_state_to_string(WTI_LIVE_RESTORE_STATE_LOG_COPY, state_to_write);
+
+        WT_ERR(__live_restore_get_state_file_path(
+          session, lr_fs->destination.home, NULL, &state_file_name, NULL));
+
+        WT_ERR(lr_fs->os_file_system->fs_open_file(lr_fs->os_file_system, (WT_SESSION *)session,
+          (char *)state_file_name->data, WT_FS_OPEN_FILE_TYPE_REGULAR,
+          WT_FS_OPEN_CREATE | WT_FS_OPEN_EXCLUSIVE, &fh));
+
+        fh->fh_write(fh, (WT_SESSION *)session, 0, 128, state_to_write);
+
+        lr_fs->state = WTI_LIVE_RESTORE_STATE_LOG_COPY;
+    }
+
+err:
+
+    __wt_writeunlock(session, &lr_fs->state_lock);
+    __wt_scr_free(session, &state_file_name);
+
+    if (fh != NULL)
+        WT_TRET(fh->close(fh, &session->iface));
+
+    return (ret);
+}
+
+/*
+ * __wti_live_restore_get_state --
+ *     Get the live restore state.
+ */
+WTI_LIVE_RESTORE_STATE
+__wti_live_restore_get_state(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FS *lr_fs)
+{
+    WTI_LIVE_RESTORE_STATE state;
+    __wt_readlock(session, &lr_fs->state_lock);
+    state = lr_fs->state;
+    __wt_readunlock(session, &lr_fs->state_lock);
+
+    /* We initialize state on startup. This shouldn't be possible. */
+    WT_ASSERT_ALWAYS(session, state != WTI_LIVE_RESTORE_STATE_NONE, "State not initialized!");
+
+    return (state);
+}
+
+/*
+ * __wti_live_restore_get_state_no_lock --
+ *     Get the live restore state without taking a lock. The caller must hold the state lock when
+ *     calling this function.
+ */
+WTI_LIVE_RESTORE_STATE
+__wti_live_restore_get_state_no_lock(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FS *lr_fs)
+{
+    WTI_LIVE_RESTORE_STATE state = lr_fs->state;
+
+    /* We initialize state on startup. This shouldn't be possible. */
+    WT_ASSERT_ALWAYS(session, state != WTI_LIVE_RESTORE_STATE_NONE, "State not initialized!");
+
```

Comment:
See comment elsewhere about having a 'locked' boolean and not having two functions. You can find an example in `os_common/os_fhandle.c:__wt_handle_is_open`.